test/unit/colorspace_spec.js

```diff
@@ -0,0 +1,541 @@
+/* Copyright 2017 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Dict, isRef, Name, Ref } from '../../src/core/primitives';
+import { Stream, StringStream } from '../../src/core/stream';
+import { ColorSpace } from '../../src/core/colorspace';
+
+describe('colorspace', function () {
+  class XRefMock {
+    constructor(array) {
+      this.map = Object.create(null);
+      for (let elem in array) {
+        let obj = array[elem];
+        let ref = obj.ref;
+        let data = obj.data;
+        this.map[ref.toString()] = data;
+      }
+    }
+
+    fetch(ref) {
+      return this.map[ref.toString()];
+    }
+
+    fetchIfRef(obj) {
+      if (!isRef(obj)) {
+        return obj;
+      }
+      return this.fetch(obj);
+    }
+  }
+
+  describe('ColorSpace', function () {
+    it('should be true if decode is not an array', function () {
+      expect(ColorSpace.isDefaultDecode('string', 0)).toBeTruthy();
+    });
+    it('should be true if length of decode array is not correct',
+        function () {
+      expect(ColorSpace.isDefaultDecode([0], 1)).toBeTruthy();
+      expect(ColorSpace.isDefaultDecode([0, 1, 0], 1)).toBeTruthy();
+    });
+    it('should be true if decode map matches the default decode map',
+        function () {
+      expect(ColorSpace.isDefaultDecode([], 0)).toBeTruthy();
+
+      expect(ColorSpace.isDefaultDecode([0, 0], 1)).toBeFalsy();
+      expect(ColorSpace.isDefaultDecode([0, 1], 1)).toBeTruthy();
+
+      expect(ColorSpace.isDefaultDecode([0, 1, 0, 1, 0, 1], 3)).toBeTruthy();
+      expect(ColorSpace.isDefaultDecode([0, 1, 0, 1, 1, 1], 3)).toBeFalsy();
+
+      expect(ColorSpace.isDefaultDecode([0, 1, 0, 1, 0, 1, 0, 1], 4))
+        .toBeTruthy();
+      expect(ColorSpace.isDefaultDecode([1, 0, 0, 1, 0, 1, 0, 1], 4))
+        .toBeFalsy();
+    });
+  });
+
+  describe('DeviceGrayCS', function () {
+    it('should handle the case when cs is a Name object', function () {
+      let cs = new Name('DeviceGray');
```

Comment:
I'm going to comment on this once here, but it applies in a number of places below as well:
Please use `Name.get(''DeviceGray')` instead, since that only creates one instance of the `Name` object regardless of how many times it's called.